lib/ecto/adapters/sql.ex

```diff
@@ -1377,21 +1425,37 @@ defmodule Ecto.Adapters.SQL do
 
   @repo_modules [Ecto.Repo.Queryable, Ecto.Repo.Schema, Ecto.Repo.Transaction]
 
-  defp last_non_ecto([{mod, _, _, _} | _stacktrace], repo, last)
+  @doc """
+  Receives a stacktrace, and return the first N items before Ecto entries
+
+  This function is used by default in the `:log_stacktrace_info` config, with
+  a size of 1.
+  """
+  @spec first_non_ecto_stacktrace(Exception.stacktrace(), Ecto.Repo.t(), non_neg_integer()) ::
+          Exception.stacktrace()
+  def first_non_ecto_stacktrace(stacktrace, repo, size) do
+    stacktrace
+    |> Enum.reverse()
+    |> last_non_ecto_entries(repo, [])
+    |> Enum.take(size)
```

Comment:
I'm not sure how big these stacktraces can get. Is it better to pass a decrementing counter to `last_non_ecto_entries` and reverse after that? Or it doesn't really matter?

Reply:
Default is 8, biggest I have ever seen was probably like 20.

I think this is __fine__, but we can give it a stab.

Reply:
@greg-rychlewski I noticed you resolved. I didn't mean to dismiss, I think you are right, I'll try to implement ASAP :)